spynnaker/pyNN/models/neuron/population_machine_vertex.py

```diff
@@ -384,6 +397,227 @@ def get_binary_file_name(self):
+
+        # close spec
+        spec.end_specification()
+
+    @overrides(AbstractRewritesDataSpecification.reload_required)
+    def reload_required(self):
+        return self.__change_requires_neuron_parameters_reload
+
+    @overrides(AbstractRewritesDataSpecification.set_reload_required)
+    def set_reload_required(self, new_value):
+        self.__change_requires_neuron_parameters_reload = new_value
+
+    def _reserve_memory_regions(self, spec, machine_graph, n_key_map):
+        """ Reserve the DSG data regions.
+
+        :param ~.DataSpecificationGenerator spec:
+            the spec to write the DSG region to
+        :param ~.MachineGraph machine_graph: machine graph
+        :param n_key_map: n key map
+        :return: None
+        """
+        spec.comment("\nReserving memory space for data regions:\n\n")
+
+        # Reserve memory:
+        spec.reserve_memory_region(
+            region=POPULATION_BASED_REGIONS.SYSTEM.value,
+            size=common_constants.SIMULATION_N_BYTES,
+            label='System')
+
+        self._reserve_neuron_params_data_region(spec)
+
+        spec.reserve_memory_region(
+            region=POPULATION_BASED_REGIONS.NEURON_RECORDING.value,
+            size=self._app_vertex.neuron_recorder.get_static_sdram_usage(
+                self.vertex_slice),
+            label="neuron recording")
+
+        profile_utils.reserve_profile_region(
+            spec, POPULATION_BASED_REGIONS.PROFILING.value,
+            self._app_vertex.n_profile_samples)
+
+        # reserve bit field region
+        bit_field_utilities.reserve_bit_field_regions(
+            spec, machine_graph, n_key_map, self,
+            POPULATION_BASED_REGIONS.BIT_FIELD_BUILDER.value,
+            POPULATION_BASED_REGIONS.BIT_FIELD_FILTER.value,
+            POPULATION_BASED_REGIONS.BIT_FIELD_KEY_MAP.value)
+
+        self.reserve_provenance_data_region(spec)
+
+    @staticmethod
+    def neuron_region_sdram_address(placement, transceiver):
+        return helpful_functions.locate_memory_region_for_placement(
+                placement, POPULATION_BASED_REGIONS.NEURON_PARAMS.value,
+                transceiver)
+
+    def _reserve_neuron_params_data_region(self, spec):
+        """ Reserve the neuron parameter data region.
+
+        :param ~data_specification.DataSpecificationGenerator spec:
+            the spec to write the DSG region to
+        :return: None
+        """
+        params_size = self._app_vertex.get_sdram_usage_for_neuron_params(
+            self.vertex_slice)
+        spec.reserve_memory_region(
+            region=POPULATION_BASED_REGIONS.NEURON_PARAMS.value,
+            size=params_size, label='NeuronParams')
+
+    def _write_neuron_parameters(self, spec, key, region_id):
+
+        self._app_vertex.update_state_variables()
+
+        # pylint: disable=too-many-arguments
+        n_atoms = self.vertex_slice.n_atoms
+        spec.comment("\nWriting Neuron Parameters for {} Neurons:\n".format(
```

Comment:
Commen needs to point to specific neuron now

Reply:
its stil talking about a number of neurons.